cmd/config.go

```diff
@@ -2,43 +2,96 @@ package cmd
 
 import (
 	"fmt"
-	"io/ioutil"
 
-	"github.com/mcuadros/go-defaults"
+	"github.com/MakeNowJust/heredoc"
 	"github.com/odpf/guardian/app"
+	"github.com/odpf/salt/cmdx"
 	"github.com/spf13/cobra"
-	"gopkg.in/yaml.v3"
 )
 
 func configCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "config",
-		Short: "manage guardian CLI configuration",
+		Use:   "config <command>",
+		Short: "Manage client configuration settings",
+		Example: heredoc.Doc(`
+			$ stencil config init
+			$ stencil config get
+			$ stencil config list`),
 	}
+
 	cmd.AddCommand(configInitCommand())
+	cmd.AddCommand(configListCommand())
+	cmd.AddCommand(configGetCommand())
+
 	return cmd
 }
 
 func configInitCommand() *cobra.Command {
 	return &cobra.Command{
 		Use:   "init",
-		Short: "initialize CLI configuration",
+		Short: "Initialize a new client configuration",
+		Example: heredoc.Doc(`
+			$ stencil init --path=.stencil.yml
+		`),
+		Annotations: map[string]string{
+			"group:core": "true",
+		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			var config app.CLIConfig
-			defaults.SetDefaults(&config)
+			cfg := cmdx.SetConfig("guardian")
+
+			if err := cfg.Init(&app.CLIConfig{}); err != nil {
+				return err
+			}
 
-			b, err := yaml.Marshal(&config)
+			fmt.Printf("config created: %v\n", cfg.File())
+			return nil
+		},
+	}
+}
+
+func configGetCommand() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "get <key>",
+		Short: "Update configuration with a value for the given key",
+		Example: heredoc.Doc(`
+			$ stencil config get --path=.stencil.yml
+		`),
+		Annotations: map[string]string{
+			"group:core": "true",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			config, err := app.LoadCLIConfig()
 			if err != nil {
 				return err
 			}
+			fmt.Println(config)
```

Comment:
this get command isn't returning one specified config key

Reply:
Removed get command for now.